pyiron_atomistics/dft/bader.py

```diff
@@ -0,0 +1,124 @@
+    `Bader code`_ from the Greame Henkelmann group.
+
+    .. _Bader code: http://theory.cm.utexas.edu/henkelman/code/bader
+    """
+    def __init__(self, job):
+        """
+        Initialize the Bader module
+
+        Args:
+            job (pyiron_atomistics.dft.job.generic.GenericDFTJob): A DFT job instance (finished/converged job)
+        """
+        self.job = job
+        self._working_directory = job.working_directory
+        self._structure = job.structure
+
+    def _create_cube_files(self):
+        """
+        Create CUBE format files of the total and valce charges to be used by the Bader program
+        """
+        cd_val, cd_total = self.job.get_valence_and_total_charge_density()
+        cd_val.write_cube_file(filename=os.path.join(self._working_directory, "valence_charge.CUBE"))
+        cd_total.write_cube_file(filename=os.path.join(self._working_directory, "total_charge.CUBE"))
+
+    def call_bader_from_job(self, extra_arguments=None):
+        """
+        Run Bader analysis on the output from the DFT job
+
+        Args:
+            extra_arguments (str): Extra arguments to the Bader program
+
+        Returns:
+            tuple: Charges and volumes as numpy arrays
+
+        """
+        self._create_cube_files()
+        error_code = call_bader(foldername=self._working_directory, extra_arguments=extra_arguments)
+        if error_code > 0:
+            self._remove_cube_files()
+            raise ValueError("Invoking Bader charge analysis failed!")
+        self._remove_cube_files()
+        return self._parse_charge_vol()
+
+    def _remove_cube_files(self):
+        """
+        Delete created CUBE files
+        """
+        os.remove(os.path.join(self._working_directory, "valence_charge.CUBE"))
+        os.remove(os.path.join(self._working_directory, "total_charge.CUBE"))
+
+    def _parse_charge_vol(self):
+        """
+        Parse Bader charges and volumes
+
+        Returns:
+            tuple: charges and volumes
+
+        """
+        filename = os.path.join(self._working_directory, "ACF.dat")
+        return parse_charge_vol_file(structure=self._structure, filename=filename)
+
+
+def call_bader(foldername, extra_arguments=None):
+    """
+    Call the Bader program inside a given folder
+
+    Args:
+        foldername (str): Folder path
+        extra_arguments (str): Extra arguments to the Bader program
+
+    Returns:
+        int: Result from the subprocess call (>0 if an error occurs)
+
+    """
+    # Go to the working directory to do this
+    cmd_1 = "cd {}".format(foldername)
+    cmd_2 = list()
+    cmd_2.append("bader")
+    cmd_2.append("{0}/valence_charge.CUBE -ref {0}/total_charge.CUBE".format(foldername))
+    if extra_arguments is not None:
+        cmd_2.append(extra_arguments)
+    cmd_2 = " ".join(cmd_2)
+    return subprocess.call(";".join([cmd_1, cmd_2]), shell=True)
```

Comment:
I guess this call can be simplified. On the one hand you can set the current working directory of the subprocess using `cwd` and on the other hand for reading the code it is confusing when lists and strings are combined. Either make each parameter a separate element in a list or build one long string, both is fine but the combination is confusing for me.

Reply:
Done!